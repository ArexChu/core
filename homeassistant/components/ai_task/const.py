"""Constants for the AI Task integration."""

from __future__ import annotations

from enum import StrEnum
from typing import TYPE_CHECKING

from homeassistant.util.hass_dict import HassKey

if TYPE_CHECKING:
    from homeassistant.helpers.entity_component import EntityComponent

    from .entity import AITaskEntity

DOMAIN = "ai_task"
DATA_COMPONENT: HassKey[EntityComponent[AITaskEntity]] = HassKey(DOMAIN)

DEFAULT_SYSTEM_PROMPT = (
    "You are a Home Assistant expert and help users with their tasks."
)


class GenTextTaskType(StrEnum):
    """Generate text task types.

    A task type describes the intent of the request in order to
    match the right model for balance of cost and quality.
    """

    GENERATE = "generate"
    """Generate content, which may target a higher quality result."""

    SUMMARY = "summary"
    """Summarize existing content, which be able to use a more cost effective model."""
