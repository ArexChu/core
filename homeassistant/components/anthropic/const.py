"""Constants for the Anthropic integration."""

import logging

DOMAIN = "anthropic"
LOGGER = logging.getLogger(__package__)

DEFAULT_CONVERSATION_NAME = "Claude conversation"

CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "claude-3-5-haiku-latest"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 3000
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
CONF_THINKING_BUDGET = "thinking_budget"
RECOMMENDED_THINKING_BUDGET = 0
MIN_THINKING_BUDGET = 1024

NON_THINKING_MODELS = [
    "claude-3-5",  # Both sonnet and haiku
    "claude-3-opus",
    "claude-3-haiku",
]
