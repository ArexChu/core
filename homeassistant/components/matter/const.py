"""Constants for the Matter integration."""

import logging

ADDON_SLUG = "core_matter_server"

CONF_INTEGRATION_CREATED_ADDON = "integration_created_addon"
CONF_USE_ADDON = "use_addon"

DOMAIN = "matter"
LOGGER = logging.getLogger(__package__)

# prefixes to identify device identifier id types
ID_TYPE_DEVICE_ID = "deviceid"
ID_TYPE_SERIAL = "serial"

FEATUREMAP_ATTRIBUTE_ID = 65532

SERVICE_SET_UNOCCUPIED_COOLING_TEMPERATURE = "set_unoccupied_cooling_target_temperature"
SERVICE_SET_UNOCCUPIED_HEATING_TEMPERATURE = "set_unoccupied_heating_target_temperature"

ATTR_OCCUPANCY = "occupancy"
ATTR_UNOCCUPIED_COOLING_TARGET_TEMP = "unoccupied_cooling_target_temp"
ATTR_UNOCCUPIED_HEATING_TARGET_TEMP = "unoccupied_heating_target_temp"
