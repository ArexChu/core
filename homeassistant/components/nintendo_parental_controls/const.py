"""Constants for the Nintendo Switch parental controls integration."""

DOMAIN = "nintendo_parental_controls"
CONF_UPDATE_INTERVAL = "update_interval"
CONF_SESSION_TOKEN = "session_token"
