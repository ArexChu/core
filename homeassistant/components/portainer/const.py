"""Constants for the Portainer integration."""

DOMAIN = "portainer"
DEFAULT_NAME = "Portainer"

ENDPOINT_STATUS_DOWN = 2
