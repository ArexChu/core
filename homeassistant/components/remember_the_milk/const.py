"""Constants for the Remember The Milk integration."""

import logging

DOMAIN = "remember_the_milk"
LOGGER = logging.getLogger(__package__)
