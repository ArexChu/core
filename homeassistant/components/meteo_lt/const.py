"""Constants for the Meteo.lt integration."""

from datetime import timedelta

from homeassistant.const import Platform

DOMAIN = "meteo_lt"
PLATFORMS = [Platform.WEATHER]

MANUFACTURER = "Lithuanian Hydrometeorological Service"
MODEL = "Weather Station"

DEFAULT_UPDATE_INTERVAL = timedelta(minutes=30)

CONF_PLACE_CODE = "place_code"

ATTRIBUTION = "Data provided by Lithuanian Hydrometeorological Service (LHMT)"
