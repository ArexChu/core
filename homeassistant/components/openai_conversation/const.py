"""Constants for the OpenAI Conversation integration."""

from dataclasses import dataclass, field
import logging
from typing import Any

from homeassistant.const import CONF_LLM_HASS_API
from homeassistant.helpers import llm

DOMAIN = "openai_conversation"
LOGGER: logging.Logger = logging.getLogger(__package__)

DEFAULT_CONVERSATION_NAME = "OpenAI Conversation"
DEFAULT_AI_TASK_NAME = "OpenAI AI Task"
DEFAULT_NAME = "OpenAI Conversation"

CONF_CHAT_MODEL = "chat_model"
CONF_CODE_INTERPRETER = "code_interpreter"
CONF_FILENAMES = "filenames"
CONF_MAX_TOKENS = "max_tokens"
CONF_PROMPT = "prompt"
CONF_REASONING_EFFORT = "reasoning_effort"
CONF_RECOMMENDED = "recommended"
CONF_TEMPERATURE = "temperature"
CONF_TOP_P = "top_p"
CONF_WEB_SEARCH = "web_search"
CONF_WEB_SEARCH_USER_LOCATION = "user_location"
CONF_WEB_SEARCH_CONTEXT_SIZE = "search_context_size"
CONF_WEB_SEARCH_CITY = "city"
CONF_WEB_SEARCH_REGION = "region"
CONF_WEB_SEARCH_COUNTRY = "country"
CONF_WEB_SEARCH_TIMEZONE = "timezone"
RECOMMENDED_CODE_INTERPRETER = False
RECOMMENDED_CHAT_MODEL = "gpt-4o-mini"
RECOMMENDED_MAX_TOKENS = 3000
RECOMMENDED_REASONING_EFFORT = "low"
RECOMMENDED_TEMPERATURE = 1.0
RECOMMENDED_TOP_P = 1.0
RECOMMENDED_WEB_SEARCH = False
RECOMMENDED_WEB_SEARCH_CONTEXT_SIZE = "medium"
RECOMMENDED_WEB_SEARCH_USER_LOCATION = False

UNSUPPORTED_MODELS: list[str] = [
    "o1-mini",
    "o1-mini-2024-09-12",
    "o1-preview",
    "o1-preview-2024-09-12",
    "gpt-4o-realtime-preview",
    "gpt-4o-realtime-preview-2024-12-17",
    "gpt-4o-realtime-preview-2024-10-01",
    "gpt-4o-mini-realtime-preview",
    "gpt-4o-mini-realtime-preview-2024-12-17",
]

UNSUPPORTED_WEB_SEARCH_MODELS: list[str] = [
    "gpt-3.5",
    "gpt-4-turbo",
    "gpt-4.1-nano",
    "o1",
    "o3-mini",
]


@dataclass(frozen=True)
class ConversationOptions:
    """Configuration options for conversation."""

    recommended: bool = True
    llm_hass_api: list[str] = field(default_factory=lambda: [llm.LLM_API_ASSIST])
    prompt: str = llm.DEFAULT_INSTRUCTIONS_PROMPT
    chat_model: str = RECOMMENDED_CHAT_MODEL
    max_tokens: int = RECOMMENDED_MAX_TOKENS
    temperature: float = RECOMMENDED_TEMPERATURE
    top_p: float = RECOMMENDED_TOP_P
    reasoning_effort: str = RECOMMENDED_REASONING_EFFORT
    web_search: bool = RECOMMENDED_WEB_SEARCH
    code_interpreter: bool = RECOMMENDED_CODE_INTERPRETER

    def to_dict(self) -> dict[str, Any]:
        """Convert to dictionary format."""
        return {
            CONF_RECOMMENDED: self.recommended,
            CONF_LLM_HASS_API: self.llm_hass_api,
            CONF_PROMPT: self.prompt,
            CONF_CHAT_MODEL: self.chat_model,
            CONF_MAX_TOKENS: self.max_tokens,
            CONF_TEMPERATURE: self.temperature,
            CONF_TOP_P: self.top_p,
            CONF_REASONING_EFFORT: self.reasoning_effort,
            CONF_WEB_SEARCH: self.web_search,
            CONF_CODE_INTERPRETER: self.code_interpreter,
        }


@dataclass(frozen=True)
class AITaskOptions:
    """Configuration options for AI tasks."""

    recommended: bool = True

    def to_dict(self) -> dict[str, Any]:
        """Convert to dictionary format."""
        return {CONF_RECOMMENDED: self.recommended}


# Maintain backward compatibility with existing dictionary format
RECOMMENDED_CONVERSATION_OPTIONS = {
    CONF_RECOMMENDED: True,
    CONF_LLM_HASS_API: [llm.LLM_API_ASSIST],
    CONF_PROMPT: llm.DEFAULT_INSTRUCTIONS_PROMPT,
}
RECOMMENDED_AI_TASK_OPTIONS = {
    CONF_RECOMMENDED: True,
}
