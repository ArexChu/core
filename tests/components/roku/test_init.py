"""Tests for the Roku integration."""

from unittest.mock import AsyncMock, MagicMock, patch

from rokuecp import RokuConnectionError
from syrupy import SnapshotAssertion

from homeassistant.components.roku.const import DOMAIN
from homeassistant.config_entries import ConfigEntryState
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr

from tests.common import MockConfigEntry


async def test_device_info(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    device_registry: dr.DeviceRegistry,
    init_integration: MockConfigEntry,
) -> None:
    """Test device registry integration."""
    device_entry = device_registry.async_get_device(
        identifiers={(DOMAIN, init_integration.unique_id)}
    )
    assert device_entry is not None
    assert device_entry == snapshot


@patch(
    "homeassistant.components.roku.coordinator.Roku._request",
    side_effect=RokuConnectionError,
)
async def test_config_entry_not_ready(
    mock_request: MagicMock, hass: HomeAssistant, mock_config_entry: MockConfigEntry
) -> None:
    """Test the Roku configuration entry not ready."""
    mock_config_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_request.call_count == 1
    assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY


async def test_config_entry_no_unique_id(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_roku: AsyncMock,
) -> None:
    """Test the Roku configuration entry with missing unique id."""
    mock_config_entry.add_to_hass(hass)
    hass.config_entries.async_update_entry(mock_config_entry, unique_id=None)
    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_config_entry.state is ConfigEntryState.LOADED


async def test_load_unload_config_entry(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_roku: AsyncMock,
) -> None:
    """Test the Roku configuration entry loading/unloading."""
    mock_config_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_config_entry.state is ConfigEntryState.LOADED

    await hass.config_entries.async_unload(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_config_entry.state is ConfigEntryState.NOT_LOADED
