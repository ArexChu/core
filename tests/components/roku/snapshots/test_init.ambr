# serializer version: 1
# name: test_device_info
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
      tuple(
        'mac',
        'b0:a7:37:96:4d:fa',
      ),
      tuple(
        'mac',
        'b0:a7:37:96:4d:fb',
      ),
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': '4200X',
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'roku',
        '1GU48T017973',
      ),
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'Roku',
    'model': 'Roku 3',
    'model_id': None,
    'name': 'My Roku 3',
    'name_by_user': None,
    'primary_config_entry': <ANY>,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': '7.5.0',
    'via_device_id': None,
  })
# ---
